src/bindings/wasm/tvgWasmLottieAnimation.cpp

```diff
@@ -84,11 +104,40 @@ class __attribute__((visibility("default"))) TvgLottieAnimation
             WGPUSurfaceDescriptor surfaceDesc{};
             surfaceDesc.nextInChain = &canvasDesc.chain;
             surface = wgpuInstanceCreateSurface(instance, &surfaceDesc);
-
-            canvas = WgCanvas::gen();
         #endif
+
+        Initializer::init(0, tvg::CanvasEngine::Wg);
+        return WgCanvas::gen();
+    }
+
+    void resize(Canvas* canvas, int w, int h) override
+    {
+        static_cast<WgCanvas*>(canvas)->target(instance, surface, w, h);
+    }
+};
+
+
+class __attribute__((visibility("default"))) TvgLottieAnimation
+{
+public:
+    ~TvgLottieAnimation()
+    {
+        delete(engine);
+    }
+
+    explicit TvgLottieAnimation(string engine = "sw", string selector = "")
+    {
+        errorMsg = NoError;
+
+        if (engine == "sw") this->engine = new TvgSwEngine;
+        else if (engine == "wg") this->engine = new TvgWgEngine;
+
+        canvas = this->engine->init(selector);
```

Comment:
just noticed:
```
if (engine) {
   canvas = this->engine->init(selector);
}

Reply:
How about this code?

```cpp
if (!engine) {
    errorMsg = "Invalid engine";
    return;
}
```

Reply:
@tinyjin Let's simplify it, We can regards canvas / engine as one.